src/content/reference/react/experimental_taintUniqueValue.md

```diff
@@ -45,30 +45,78 @@ Call `taintUniqueValue` with a password, token, key or hash to register it with
 import {experimental_taintUniqueValue} from 'react';
 
 experimental_taintUniqueValue(
-  'Do not pass super secret keys to the client.',
+  'Do not pass secret keys to the client.',
   process,
-  process.env.SUPER_SECRET_KEY
+  process.env.SECRET_KEY
 );
 ```
 
 [See more examples below.](#usage)
 
 #### Parameters {/*parameters*/}
 
-* `message`: The message you want to display if the object gets passed to a Client Component. It's logged as any other Error that is thrown.
+* `errMessage`: The message you want to display if the object gets passed to a Client Component. If `value` is passed to a Client Component an Error will be thrown. `errMessage` let's you define the message displayed by this Error.
 
-* `lifetime`: Any object that indicates how long to block it. The value remains blocked as long as this object is still around. For example, passing `globalThis` blocks the value for the life time of an app. Typically it would an object whose properties contains the value.
+* `lifetime`: Any object that indicates how long `value` should be tainted. `value` will be blocked from being sent to any Client Component while this object still exists. For example, passing `globalThis` blocks the value for the lifetime of an app. `lifetime` is typically an object whose properties contains `value`.
 
-* `value`: A string, bigint or TypedArray. This value must be a unique sequence of characters/bytes with high entropy such as a cryptographic token, private key, hash or a long password. This value gets banned for passing to any Client Component in the whole app.
+* `value`: A string, bigint or TypedArray. `value` must be a unique sequence of characters or bytes with high entropy such as a cryptographic token, private key, hash, or a long password. `value` will be blocked from being sent to any Client Component.
 
 #### Returns {/*returns*/}
 
 `experimental_taintUniqueValue` returns `undefined`.
 
+#### Caveats {/*caveats*/}
+
+- Modifying tainted values can invalidate tainting protection. Converting tainted values to upper case, concatenating tainted string values into a larger string, converting tainted values to base64, returning a substring, and similar transformations will lose tainting protection.
```

Comment:
Not sure "modifying" is the right word here because you can't really mutate them which might be implied. It's more about deriving into new values than modifying the existing ones. E.g. concatenation isn't really modifying.